katana.py

```diff
@@ -0,0 +1,21 @@
+import sys
+import collections
+
+from core.notify import Notify
+
+def main():
+    arg_names = ['title', 'message', 'app_name', 'app_icon']
+
+    args = collections.namedtuple('data', arg_names)(
+        *(dict(zip(arg_names, sys.argv[1:])).get(arg, "") for arg in arg_names))
+
+    TITLE = args.title if args.title else "Katana"
+    MESSAGE = args.message if args.message else "Hello World from Katana"
+    APP_NAME = args.app_name if args.app_name else "Katana"
+    APP_ICON = args.app_name if args.app_name else "./assets/katana-logo.svg"
+
+    Notify().doNotify(TITLE, MESSAGE, APP_NAME, APP_ICON)
```

Comment:
Nome todo em maiúsculo não deveria ser somente para constantes?

Reply:
Nesse caso o `Notify()` é a classe e o `doNotify()` é o método.

Reply:
Me referia ao `TITLE`, `MESSAGE`... :P

Reply:
aah, então cara. Na verdade eu queria tratar esses caras como algo imutável dentro do código. tem algum padrão pra isso ?

Reply:
Não sei ao certo, na verdade não entendi o motivo de tornar imutável. Pensando aqui acredito que você poderia tratar de outras formas:

- Alimentar o `args` com os parâmetros informados pela chamada.
- Depois passar validando se existe alguma que não foi preenchida e ai então definir o valor default

ou 

- Criar o `args` com os valores default já
- Atualizar os valores dele usando os passados pela chamada

PS: Sobre constantes pelo que vi não é algo que o Python suporte, mas é possível utilizar alguns mecanismos para isso, da uma olhada na ideia do [Artem Nepo neste post aqui,](https://www.it-swarm.dev/pt/python/qual-e-melhor-maneira-de-definir-variaveis-constantes-python-3/838259896/) acho que é a terceira resposta.

Reply:
então, não usei o próprio `args`, porque o a chamada do método iria ficar muito extensa.
Apos a pesquisa das classes, encontrei algo interessante. Vou tentar aplicar aqui vamos ver como fica.